source/visionEnhancementProviders/screenCurtain.py

```diff
@@ -0,0 +1,84 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2018-2019 NV Access Limited, Babbage B.V., Leonard de Ruijter
+
+"""Screen curtain implementation based on the windows magnification API.
+This implementation only works on Windows 8 and above.
+"""
+
+import vision
+import winVersion
+from ctypes import Structure, windll, c_float, POINTER, WINFUNCTYPE, WinError
+from ctypes.wintypes import BOOL
+
+
+class MAGCOLOREFFECT(Structure):
+	_fields_ = (("transform", c_float * 5 * 5),)
+
+
+TRANSFORM_BLACK = MAGCOLOREFFECT()
+TRANSFORM_BLACK.transform[4][4] = 1.0
+
+
+def _errCheck(result, func, args):
+	if result == 0:
+		raise WinError()
+	return args
+
+
+class Magnification:
+	"""Singleton that wraps necessary functions from the Windows magnification API."""

```

Comment:
I think calling this a singleton is misleading, since it doesn't follow the design pattern. This seems to be more like a static wrapper class for a set of windows API functions.

Reply:
Yes, this is True. Actually we could as well store this in a separate module.

Reply:
@feerrenrut: Do you want this in a separate module, or should I leave it as is?

Reply:
Happy for it to stay as is. Though if we need to use it from somewhere else it should be moved.